### DISCUSSION TITLE
Can't find layouts when not in default location?

### DISCUSSION THREAD

**Discussion by jg3** [11ty/eleventy#2961] [Q&A] [answered]
Hi, I'm new.  New here, new to 11ty, new to static site builders, templates, etc. etc.  ... so I'm likely missing something totally obvious.  I'm using 11ty v2.0.1 ...

I'm getting this output/error (I'm not sure why it is referencing `_includes`)
```
?1 11ty/NOTES % npx @11ty/eleventy                                                        
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] TemplateLayoutPathResolver directory does not exist for layouts/base.njk: _includes (via Error)
...
```

while my eleventy.config.js file looks like this:
```
module.exports = function(eleventyConfig) {
  eleventyConfig.addPassthroughCopy("NOTES/includes/");
  return {
    dir: {
      input: "NOTES",
      output: "docs",
      // I don't know why this isn't working
      layouts: "layouts",
      includes: "includes"
    },
    pathPrefix: "pages/jamegill/notes/"
  }
};
```

It was working fine before I moved layouts outside of the includes directory, but I decided I didn't want to include those files in the stuff that gets pass-through copied up.

Am I missing something?

My directory structure looks like this (includes and layouts are at the same level, directly under NOTES, the input directory):
```
jg3 ~/11ty % tree                               
   |-NOTES
   |---includes
   |---layouts
   |-docs
   |---11ty-notes
   |---includes
   |-node_modules
   |---@11ty
...
```

**Comment by pdehaan** [comment]
https://www.11ty.dev/docs/config/#directory-for-layouts-(optional)
I believe the ‘layouts’ folder is relative to the ‘includes’ folder, but the wording in the docs is a bit contradictory.

**Reply by pdehaan** [reply, marked as answer]
Actually, I think this is working for me w/ 11ty v2.0.1, https://github.com/pdehaan/11ty-2961

```sh
npm run build

> 11ty-2961@1.0.0 build
> eleventy

[11ty] Writing docs/index.html from ./NOTES/index.njk
[11ty] Copied 1 file / Wrote 1 file in 0.03 seconds (v2.0.1)


tree -aI node_modules
.
├── NOTES
│   ├── includes
│   │   └── about.njk
│   ├── index.njk
│   └── layouts
│       └── base.njk
├── docs
│   ├── includes
│   │   └── about.njk
│   └── index.html
├── eleventy.config.js
├── package-lock.json
└── package.json

5 directories, 8 files
```

```js
/**
 * @param {import("@11ty/eleventy/src/UserConfig")} eleventyConfig
 * @returns {ReturnType<import("@11ty/eleventy/src/defaultConfig")>}
 */
module.exports = function (eleventyConfig) {
  eleventyConfig.addPassthroughCopy("NOTES/includes");
  return {
    dir: {
      input: "NOTES",
      includes: "includes",
      layouts: "layouts",
      output: "docs",
    },
  };
};
```

**Comment by jg3** [comment]
Ok, thanks you for taking the time ... I don't know what I'm doing wrong/differently here I'll make use of the little starter you made.  I appreciate it.

**Reply by pdehaan** [reply]
Are you on 2.0.1 or an older version?

**Reply by jg3** [reply]
I'm on v2.0.1 ... and I figured it out.

The error message I was seeing: 
 
`TemplateLayoutPathResolver directory does not exist for layouts/base.njk` 

... was because in the front matter of an `.md` content file I had a _layout_ line like this:

```md
---
title: this page is awesome
layout:  layouts/base.njk
... (other front matter)
---
```
... see how that page is referencing `layouts/base.njk` exactly?  Because `eleventy.config.js` has the `layouts: "layouts"` line, 11ty is going to look for any "layout" referenced there in `<input_dir>/<layouts>/<whatever_was_referenced_in_the_layout_line_in_the_frontmatter>`.  So, with the configuration above it was actually looking for `NOTES/layouts/layouts/base.njk` which did not exist.

I learned a couple of things:
- It worked when I removed the path from the `layout:` statement in the front matter
- I don't need the file extension (`.njk`) there either. 
- the error message included the reference to `_includes` because I copied that from a run where I had the layouts definition line commented out in the eleventy.config file, so it was using the default.

```md
---
title: this page is actually awesome
layout: base
... (other front matter)
---
```

Thanks again for being there to help.  I'm sure to be back with more newb questions soon.

**Comment by pdehaan** [comment]
Re: _"I don't need the file extension (`.njk`) there either."_

https://www.11ty.dev/docs/layouts/#omitting-the-layouts-file-extension

<i>

> Omitting the layout file extension (for example `layout: mylayout`) causes Eleventy to cycle through all of the supported template formats (`mylayout.*`) to look for a matching layout file. There are a few drawbacks to this approach:
> 1. It is slower! Including the file extension bypasses the file search.
> 1. It is ambiguous if you have multiple layout files with the same name and different extensions (e.g. `mylayout.njk` and `mylayout.liquid`).

</i>

So be careful if you opt for the extensionless layout approach. The default behavior might change in the future and "break" your layouts; or require you to use [layout aliases](https://www.11ty.dev/docs/layouts/#layout-aliasing).

> You can disable extensionless layouts in your project with the `setLayoutResolution` Configuration API method [Added in v2.0.0] (we may swap this to be the default in a future major version of Eleventy):

**Comment by jg3** [comment]
Good point!  Thanks for pointing that out.
